src/printer.ts

```diff
@@ -188,13 +188,17 @@ export class PugPrinter {
 						// TODO: These tokens read the length of the result
 						this.result = results.join('');
 					// eslint-disable-next-line no-fallthrough
-					default:
+					default: {
+						if (typeof this[token.type] !== 'function') {
+							throw new Error('Unhandled token: ' + JSON.stringify(token));
+						}
 						// @ts-expect-error
 						results.push(this[token.type](token));
 						break;
+					}
 				}
-			} catch {
-				throw new Error('Unhandled token: ' + JSON.stringify(token));
+			} catch (error) {
+				throw new Error(error);
 			}
```

Comment:
Why catch just to throw again? You may just not catch at all?

Reply:
But it's good this part changes, this used to swallow errors, which has caused some problems for me before also…

Reply:
Yes, just didn't updated this part on a huge rewrite. It was useful some time
I re-throwed it, because in this way I can click on the code line in stack trace and come to this line